packages/attention/src/component.tsx

```diff
@@ -176,24 +200,72 @@ export function Attention(props: AttentionProps) {
   const defaultAriaLabel = () => {
     return `${activeAttentionType} ${!props.noArrow ? pointingAtDirection : ''}`
   }
-  // Recompute on re-render
+  const arrowDirectionClassname = (dir: string) => {
+    let direction: string
+    if (/-/.test(dir)) {
+      direction = dir
+        .split('-')
+        .map((d) => d.charAt(0).toUpperCase() + d.slice(1))
+        .join('')
+    } else {
+      direction = dir.charAt(0).toUpperCase() + dir.slice(1)
+    }
+    return `arrowDirection${direction}`
+  }
+  
   useEffect(() => {
     recompute(attentionState)
-  })
-
+  }, [attentionState])
+    
   useEffect(() => {
     if (isMounted.current) {
       isMounted.current = false
-
       // update attention's visibility after first render if showing by default or it's of type callout
-      if (isShowing === true || props.callout) {
+      if (isShowing || props.callout) {
         setIsVisible(isShowing)
       }
     } else {
       setIsVisible(isShowing)
     }
   }, [isShowing, props.callout])
+  
+  // @ts-ignore
+  useEffect(() => {
+    if (isShowing && flip && targetEl && attentionEl) {
+        // starts the autoUpdate, making sure the attention elements's position stays anchored to the target element 
+       const cleanup = autoUpdatePosition(attentionState);
 
+       return cleanup
+    } 
+  }, [targetEl, isShowing, attentionEl, flip])
+  
+  
+  
+
+  
+
+  const Arrow = forwardRef<HTMLDivElement, ArrowProps>(
```

Comment:
Was it necessary to move `Arrow` component and `arrowDirectionClassname` inside the Attention component scope?

Reply:
Yes! If I remember correctly, I noticed that the arrow's ref was not being set when it was outside of the attention element.

Reply:
Hmm If I remember correctly, I moved them into the attention component because I had some issues with the arrowRef not updating correctly. But I see now that it seems to work fine also being outside of the attention component, so I can revert this move! :)

Reply:
Reverted moving in `Arrow` and have also removed `arrowDirectionClassname` and use instead the `arrowDirectionClassname` that is coming from the core-library. Updated in this commit: [d147451](https://github.com/warp-ds/react/pull/215/commits/d1474513cb2bcb3efc7dc78c9c56d9fea6ede062)